src/udev/udev-builtin-input_id.c

```diff
@@ -202,20 +207,14 @@ static bool test_pointers(struct udev_device *dev,
                 has_mt_coordinates = false;
         is_direct = test_bit(INPUT_PROP_DIRECT, bitmask_props);
         has_touch = test_bit(BTN_TOUCH, bitmask_key);
-        /* joysticks don't necessarily have buttons; e. g.
-         * rudders/pedals are joystick-like, but buttonless; they have
-         * other fancy axes */
-        has_joystick_axes_or_buttons = test_bit(BTN_TRIGGER, bitmask_key) ||
-                                       test_bit(BTN_A, bitmask_key) ||
-                                       test_bit(BTN_1, bitmask_key) ||
-                                       test_bit(ABS_RX, bitmask_abs) ||
-                                       test_bit(ABS_RY, bitmask_abs) ||
-                                       test_bit(ABS_RZ, bitmask_abs) ||
-                                       test_bit(ABS_THROTTLE, bitmask_abs) ||
-                                       test_bit(ABS_RUDDER, bitmask_abs) ||
-                                       test_bit(ABS_WHEEL, bitmask_abs) ||
-                                       test_bit(ABS_GAS, bitmask_abs) ||
-                                       test_bit(ABS_BRAKE, bitmask_abs);
+
+        /* Joysticks don't necessarily have buttons; e. g.
+         * rudders/pedals are joystick-like, but buttonless;
+         * they have other fancy axes.
+         * Set has_joystick_buttons if any button in [BTN_JOYSTICK, BTN_DIGI) is found
+         * and has_joystick_axes if any axis in [ABS_RX, ABS_PRESSURE) is found */
+        for (button = BTN_JOYSTICK; button < BTN_DIGI && !(has_joystick_buttons = test_bit(button, bitmask_key)); button++);
+        for (axis = ABS_RX; axis < ABS_PRESSURE && !(has_joystick_axes = test_bit(axis, bitmask_abs)); axis++);
```

Comment:
the same here.

The old code I could understand, the new code looks like Perl.